src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,645 @@
+ */
+static constexpr Property<std::string, PropertyMutability::RO> architecture{"DEVICE_ARCHITECTURE"};
+
+/**
+ * @brief Enum to define possible device types
+ */
+enum class Type {
+    INTEGRATED = 0,  //!<  Device is integrated into host system
+    DISCRETE = 1,    //!<  Device is not integrated into host system
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const Type& device_type) {
+    switch (device_type) {
+    case Type::DISCRETE:
+        return os << "discrete";
+    case Type::INTEGRATED:
+        return os << "integrated";
+    default:
+        throw ov::Exception{"Unsupported device type"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, Type& device_type) {
+    std::string str;
+    is >> str;
+    if (str == "discrete") {
+        device_type = Type::DISCRETE;
+    } else if (str == "integrated") {
+        device_type = Type::INTEGRATED;
+    } else {
+        throw ov::Exception{"Unsupported device type: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief Read-only property to get a type of device. See Type enum definition for possible return values
+ */
+static constexpr Property<Type, PropertyMutability::RO> type{"DEVICE_TYPE"};
+
+/**
+ * @brief Read-only property which defines Giga OPS per second count (GFLOPS or GIOPS) for a set of precisions supported
+ * by specified device
+ */
+static constexpr Property<std::map<ie::Precision, float>, PropertyMutability::RO> gops{"DEVICE_GOPS"};
+
+/**
+ * @brief  Read-only property to get a float of device thermal
+ */
+static constexpr Property<float, PropertyMutability::RO> thermal{"DEVICE_THERMAL"};
+
+}  // namespace device
+
+/**
+ * @brief The key with the list of device targets used to fallback unsupported layers
+ * by HETERO plugin
+ */
+static constexpr device::Priorities target_fallback{"TARGET_FALLBACK"};
+
+/**
+ * @brief The key for enabling of dumping the topology with details of layers and details how
+ * this network would be executed on different devices to the disk in GraphViz format.
+ */
+static constexpr Property<bool, PropertyMutability::RW> dump_graph_dot{"HETERO_DUMP_GRAPH_DOT"};
+
+namespace execution {
+namespace streams {
+/**
+ * @brief Special value for ov::execution::streams::num property.
+ * Creates bare minimum of streams to improve the performance
+ */
+static constexpr const int32_t AUTO = -1;
+/**
+ * @brief Special value for ov::execution::streams::num property.
+ * Creates as many streams as needed to accommodate NUMA and avoid associated penalties
+ */
+static constexpr const int32_t NUMA = -2;
+
+/**
+ * @brief The number of executor logical partitions
+ */
+static constexpr Property<int32_t, PropertyMutability::RW> num{"NUM_STREAMS"};
+}  // namespace streams
+
+/**
+ * @brief Maximum number of concurent tasks executed by execututor
+ */
+static constexpr Property<int32_t, PropertyMutability::RW> concurrency{"CONCURRENCY"};
```

Comment:
=> compile_num_threads, inference_num_threads ?

Reply:
let's add documentation that compile_num_threads and inference_num_threads  are not intersected things